pkg/kubecrypto/certs.go

```diff
@@ -0,0 +1,257 @@
+package kubecrypto
+
+import (
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	ocrypto "github.com/scylladb/scylla-operator/pkg/crypto"
+	"github.com/scylladb/scylla-operator/pkg/helpers"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+)
+
+func needsRefresh(existingCert *x509.Certificate, now time.Time, refresh time.Duration, desiredCert *x509.Certificate, issuerPublicKey *rsa.PublicKey) string {
+	// Don't check notBefore to avoid issues on time skew.
+	// notAfter is fine as the cert should never be close to it.
+	if now.After(existingCert.NotAfter) {
+		return "already expired"
+	}
+
+	validity := existingCert.NotAfter.Sub(existingCert.NotBefore)
+	at80Percent := existingCert.NotAfter.Add(-validity / 5)
+	if now.After(at80Percent) {
+		return fmt.Sprintf("past its latest possible refresh time %v", at80Percent)
+	}
```

Comment:
where would you like it to be? (the durations are not exposed to users)

Reply:
I know it's not exposed to user, i thought it may be interesting to devs that even though they pass refresh time at 90% of validity it will be refreshed a little earlier. Maybe in ManageCertificates

Reply:
done